src/store/actions/favoritesActions.js

```diff
@@ -8,6 +18,46 @@ import {
 export const toggleIsFavoritePlaying = (payload) => ({
   type: FAVORITES_SET_IS_FAVORITE_PLAYING, payload,
 });
+
 export const removeFavorite = (payload) => ({ type: FAVORITES_REMOVE_FAVORITE, payload });
 export const addFavorite = (payload) => ({ type: FAVORITES_ADD_FAVORITE, payload });
 export const initializeFavorites = (payload) => ({ type: FAVORITES_INITIALIZE_FAVORITES, payload });
+
+export const createFavorite = (fileName, path) => (dispatch, getState) => {
+  const token = getState().app.get('token');
+  const favorites = getState().favorites.get('songs');
+
+  const body = {
+    song_name: fileName,
+    path_lower: path,
+  };
+
+  if (!isSongInFavorites(favorites, path)) {
+    createFavoriteService(token, body)
+      .then(({ data }) => {
+        const songData = Map(data);
+        dispatch(addFavorite(songData));
+      })
+      .catch(() => {
```

Comment:
All catches should report the error to rollbar (already implemented in the error logger).